File: qiniulib/src/main/java/com/shuai/qiniulib/lib/QiNiuUploadTokenLoader.java
package com.shuai.qiniulib.lib;

/**
 * 上传Token生成器接口
 */
public interface QiNiuUploadTokenLoader {

    void getUploadToken(QiNiuUploadTokenResult result);

    interface QiNiuUploadTokenResult {

        void onSuccess(String token);

        void onError(String error);
    }
}

File: qiniulib/src/main/java/com/shuai/qiniulib/lib/QiNiuUploadTokenLoader.kt
package com.shuai.qiniulib.lib

/**
 * 上传Token生成器接口
 */
interface QiNiuUploadTokenLoader {
    fun getUploadToken(result: QiNiuUploadTokenResult)
    interface QiNiuUploadTokenResult {
        fun onSuccess(token: String?)
        fun onError(error: String?)
    }
}